### DISCUSSION TITLE
BirdNET on Other Platforms?

### DISCUSSION THREAD

**Discussion by Rochambeaux** [mcguirepr89/BirdNET-Pi#435] [Q&A]
Does it exist?  Luckily I was able to re-purpose an existing Pi 4 for my initial foray into this amazing project.  Id' really like to deploy a couple units to field biology stations in the Peruvian Amazon but finding Pi's at non-felony pricing is impossible.  I have a couple of 4-5 year old Windows laptops and was recently looking at Pi alternatives like Orange Pi.  

Any hope for this ever running on Windows, Android or other LINUX builds?

**Comment by Clumpton** [comment]
Hi,
I have just successfully installed Birdnet-Pi on a Libre Computer AML-S905X-CC (LePotato).
I used Armbian Bullseye - which has the right flavour of Python - and ran the standard Birdnet-Pi install script. Worked perfectly first time.
Perfomance-wise the LePotato slots in between the Pi3 B+ and the Pi4. No built-in WiFi though.
Hope this helps.

**Reply by mcguirepr89** [reply]
Thanks, @Clumpton!!! Awesome to know!!! I did as much as I could to make the system hardware agnostic (aside from the CPU needing the 64-bit instruction set) so long as the system is on a debian-based OS -- thank you so much for taking the time to test this out!

**Reply by Clumpton** [reply]
Flushed with my success I have ordered a NanoPi NEO 512MB from China (32€ inc p&p) - this supposedly runs the same flavours of Armbian as the LePotato. It has its own audio input so it could be a solution to USB microphone woes. Expected mid-August, watch this space!

**Reply by Rochambeaux** [reply]
> Thanks, @Clumpton!!! Awesome to know!!! I did as much as I could to make the system hardware agnostic (aside from the CPU needing the 64-bit instruction set) so long as the system is on a debian-based OS -- thank you so much for taking the time to test this out!

Can you take a peek at the issue below please? I'm trying to install on an Orange Pi and keep running into dead ends.

**Comment by Phaeton** [comment]
See #444. @zgap118 had success with Orange Pi Zero 2.

**Reply by Rochambeaux** [reply]
What flavor OS did you use?

**Reply by Phaeton** [reply]
They mentioned Debian

**Reply by zgap118** [reply]
`orangepizero2:~/BirdNET-Pi$ cat /etc/debian_version`

`11.3`

**Reply by Rochambeaux** [reply]
So just got an Orange Pi 4 LTS in and struggling with which image to try:

![image](https://user-images.githubusercontent.com/86641453/180624991-2ffc1014-e0e1-40ec-bb27-1fd586c5b98a.png)

I tried Buster but it exited out, requiring python3 to be installed.  After installing it the installation script wouldn't run again (error saying directory already exists).  So I reflashed the SD Card, installed Python3 first, ran the script and it exited at: tftlite_runtime-2.6.0-cp39-none-linux-aarch64.whl is not a supported wheel on this platform.

**Reply by Phaeton** [reply]
Take a look at #444. They used Bullseye and installed sqlite and python3-venv.

**Reply by Rochambeaux** [reply]
> Take a look at #444. They used Bullseye and installed sqlite and python3-venv.

Reflashed with Bullseye, install sqlite and python3-venv.

Process went much further but eventually tanked out at:

error command '.usr/bin/aarch64-linux-ngu-gcc' failed with exit code 1
Could not build psutil which use PEP 517 and cannot be installed directly...

**Comment by cam-el** [comment]
I have it installed on a [Station P2](https://www.indiegogo.com/projects/station-p2-powerful-open-source-geek-computer#/) which uses a Rockchip 3568, ARM A55 cores, 4GB RAM and SATA3 SSD.
I also am using a bullseye [Armbian](https://www.armbian.com/station-p2/#kernels-archive-all) image from the SD flash card. I cannot get this image flavour to run the "install to SSD" tool successfully so only have SD flash or EMMC storage atm. I have used it with SATA SSD and openwrt with Frigate running from an SATA SSD + some desktop versions so I am working on the SATA issue.

After updating and configuring Armbian/IP/user and running the standard Birdnet-Pi script I had some install issues. Each time ALOT of packages are installed so run `rm -rf /home/birdnet/BirdNET-Pi` to rerun the install script and it gets further each retry. 
Had to run `sudo apt-get install python3-venv` separately to fix a dependency.
When the web interface came up I had no sound input with a known good but lousy USB sound card + microphone combo. Had to add the user the birdnet-pi installer is run under to the `audio` local group. To test run `arecord -L` with and without sudo. The output should be the same.

I have a Raspberry Pi 0W2 with a good USB and Mic setup so I have something to compare against.
On the Pi OW2 I have to disable all the services recommended in the Pi OW2 install guide and don't use the web interface to make it at all stable and even then it falls hours behind on analysing files and rops the wifi connection and/or becomes uncontactable. The 2Ghz A55's and 4GB of RAM make the Station P2 a much, much better machine to run the Birdnet-pi app and interface however I am still testing and getting some poor results. 
With the same audio setup as used on my working birdweather Pi 0W2 I am not getting good detection rates at all. I have used this [guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card)  and appear to be getting good recordings however very few detection. I have watched `journalctl -F` and the web log interface but don't see anything helpful. I will see.


edit:
uname -a 
Pi0W2
`Linux birdnetpi1 5.15.32-v8+ #1538 SMP PREEMPT Thu Mar 31 19:40:39 BST 2022 aarch64 GNU/Linux`
StationPC
`Linux station-p2 5.18.0-media #22.05.1 SMP PREEMPT Sat May 28 07:50:27 UTC 2022 aarch64 GNU/Linux`
`Python 3.9.2`

**Comment by Rochambeaux** [comment]
Any suggestions on what to try next?  Attempting to load on an Orange Pi 4 LTS:


Reflashed with Bullseye, installed sqlite and python3-venv before kicking of BirdNET installation.

Process went much further but eventually tanked out at:

error command '.usr/bin/aarch64-linux-ngu-gcc' failed with exit code 1
Could not build psutil which use PEP 517 and cannot be installed directly...

**Comment by spel-ux** [comment]
I'm definitely interested in using BirdNET on a small, lower cost 64bit SBC.

Le Potato	(Amlogic S905X SoC)
Pine Rock64 (Rockchip RK3328)
Orange Pi Zero 2 (Allwinner H616 64-bit high-performance Quad-core Cortex-A53)
NanoPi A64 (Allwinner A64, 64-bit Quad-core Cortex-A53)
PINE64 (ARM Cortex A53)

NanoPi NEO (Allwinner H3, Quad-core Cortex-A7) 32Bit   ;-(

In any case, it looks like everything is going well with Armbian!

Do you have any tips for a very small, energy-saving and at the same time inexpensive solution which is currently also available?

Thanks!

**Reply by spel-ux** [reply]
PINE64 (ARM Cortex A53) works.

**Comment by matthew73210** [comment]
Hey guys,

Just moved from a pi 3 to orange pi 4 lts, made the initial mistake of installing the latest version on armbian before installing bullseye. 
Here's what I did :  
Got wifi to work through nmtui.
Through ssh ran the install script and left it at that.
Installed in one fell swoop, no errors.

Fist thing that's different is that it didn't change the hostname, can't remember if I did it on the last rpi.
Sometimes it stopped recording and I need to go into the web terminal. I've also noticed that arecord is offline.

Cheers

**Reply by Rochambeaux** [reply]
> Hey guys,
> 
> Just moved from a pi 3 to orange pi 4 lts, made the initial mistake of installing the latest version on armbian before installing bullseye. Here's what I did : Got wifi to work through nmtui. Through ssh ran the install script and left it at that. Installed in one fell swoop, no errors.
> 
> Fist thing that's different is that it didn't change the hostname, can't remember if I did it on the last rpi. Sometimes it stopped recording and I need to go into the web terminal. I've also noticed that arecord is offline.
> 
> Cheers

What version of bullseye did you use?

I've run into numerous dead stops trying to install BirdNet on a Pi 4 LTS.

**Reply by matthew73210** [reply]
On this webpage [Armbian](https://www.armbian.com/orange-pi-4-lts/) in the _Other download variants_ I grabbed the Bullseye variant. The only issue I've had so far is that arecord looses access to the microphone when I'm logged out, probs a permission issue. However i've mitigated that by using the old pi as an rtsp server, suits my setup better. Device keeps itself a lot cooler.

May I ask what deadstops you are talking about?

**Reply by Rochambeaux** [reply]
> On this webpage [Armbian](https://www.armbian.com/orange-pi-4-lts/) in the _Other download variants_ I grabbed the Bullseye variant. The only issue I've had so far is that arecord looses access to the microphone when I'm logged out, probs a permission issue. However i've mitigated that by using the old pi as an rtsp server, suits my setup better. Device keeps itself a lot cooler.
> 
> May I ask what deadstops you are talking about?

I've been using the builds directly from the Orange Pi site so maybe the ones on the Armbian site are better suited,  

Here's where I am with my attempts on the 4 LTS:

Reflashed with Bullseye, installed sqlite and python3-venv before kicking of BirdNET installation.

Process went much further but eventually tanked out at:

error command '.usr/bin/aarch64-linux-ngu-gcc' failed with exit code 1
Could not build psutil which use PEP 517 and cannot be installed directly...

**Reply by matthew73210** [reply]
Here's exaxtly what I did.
Intsalled bullseye, set up my own account. Then ran the installer script, didn't even do an update before that. It asked for my password once or twice. Did it ask for yours ?

**Reply by Rochambeaux** [reply]
> Here's exaxtly what I did. Intsalled bullseye, set up my own account. Then ran the installer script, didn't even do an update before that. It asked for my password once or twice. Did it ask for yours ?

By "setup my own account" what did you use?  I don;t recall ti asking for PWD from the script but all the other steps I had to take to get missing stuff in did.

**Reply by matthew73210** [reply]
> 

root password and user account. The script needs a password to install packages as it refuses to run under sudo. It asks you near the start if I'm not mistaken.
